### PR TITLE
Adding solution of hashes exercise

### PR DESCRIPTION
**Pull request by tixel**
Adding hashes exercise to solution branch.
- added 1.hashes folder with basic exercise
- moved 1.elements to 3.elements (already making room for 2.headers exercises)

Thanks @guillemcordoba for fixing the exercise

### CODE REVIEW

File: basic/1.hashes/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# 1. Entries Exercise
```
Comment (guillemcordoba) [inline basic/1.hashes/README.md:1]: This should be changed to hashes right?

File: basic/1.hashes/zomes/exercise/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    Ok(WrappedEntryHash(x))
+}
+
+#[hdk_extern]
+pub fn get_book(hash: WrappedEntryHash) -> ExternResult<Book> {
+    let element:Element = get(hash.0, GetOptions::default())?.ok_or(WasmError::Guest(String::from("Could not find book")))?;
+    let bookoption: Option<Book> = element.entry().to_app_option()?;
+    let book: Book = bookoption.unwrap();
+    Ok(book) //.ok_or(WasmError::Zome(String::from("Could not convert book")))
```
Comment (guillemcordoba) [inline basic/1.hashes/zomes/exercise/src/lib.rs:41]: Maybe remove the comment?

**Review by guillemcordoba — COMMENTED** [review]
Minor things, once those are fixed I will do a pass with a couple of updates on libraries and stuff.

Thanks :D

File: basic/1.hashes/README.md (hunk 1 above)
Comment (tixel) [reply basic/1.hashes/README.md]: fixed all references to entries exercise in README

File: basic/1.hashes/zomes/exercise/src/lib.rs (hunk 2 above)
Comment (tixel) [reply basic/1.hashes/zomes/exercise/src/lib.rs]: done

**Pull request merged**